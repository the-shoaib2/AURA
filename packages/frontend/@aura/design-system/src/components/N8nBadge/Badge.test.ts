import { render } from '@testing-library/vue';

import auraBadge from './Badge.vue';

describe('components', () => {
	describe('auraBadge', () => {
		describe('props', () => {
			it('should render default theme correctly', () => {
				const wrapper = render(auraBadge, {
					props: {
						theme: 'default',
						size: 'large',
						bold: true,
					},
					slots: {
						default: '<aura-text>Default badge</aura-text>',
					},
					global: {
						stubs: ['aura-text'],
					},
				});
				expect(wrapper.html()).toMatchSnapshot();
			});
			it('should render secondary theme correctly', () => {
				const wrapper = render(auraBadge, {
					props: {
						theme: 'secondary',
						size: 'medium',
						bold: false,
					},
					slots: {
						default: '<aura-text>Secondary badge</aura-text>',
					},
					global: {
						stubs: ['aura-text'],
					},
				});
				expect(wrapper.html()).toMatchSnapshot();
			});
			it('should render with default values correctly', () => {
				const wrapper = render(auraBadge, {
					slots: {
						default: '<aura-text>A Badge</aura-text>',
					},
					global: {
						stubs: ['aura-text'],
					},
				});
				expect(wrapper.html()).toMatchSnapshot();
			});
		});
	});
});
