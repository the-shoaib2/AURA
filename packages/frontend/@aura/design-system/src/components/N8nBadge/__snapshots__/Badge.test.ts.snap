// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > auraBadge > props > should render default theme correctly 1`] = `"<span class="aura-badge default border"><aura-text-stub bold="true" size="large" compact="true" tag="span"></aura-text-stub></span>"`;

exports[`components > auraBadge > props > should render secondary theme correctly 1`] = `"<span class="aura-badge secondary border"><aura-text-stub bold="false" size="medium" compact="true" tag="span"></aura-text-stub></span>"`;

exports[`components > auraBadge > props > should render with default values correctly 1`] = `"<span class="aura-badge default border"><aura-text-stub bold="false" size="small" compact="true" tag="span"></aura-text-stub></span>"`;
